agenthub/planner_agent/agent.py

```diff
@@ -38,12 +39,12 @@ def step(self, state: State) -> Action:
         if state.plan.task.state in ['completed', 'verified', 'abandoned']:
             return AgentFinishAction()
         prompt = get_prompt(state.plan, state.history)
-        messages = [{"content": prompt, "role": "user"}]
+        messages = [{'content': prompt, 'role': 'user'}]
         resp = self.llm.completion(messages=messages)
         action_resp = resp['choices'][0]['message']['content']
+        state.num_of_chars += len(prompt) + len(action_resp)
```

Comment:
🤔 we should probably put this _outside_ of the agent, so that it only gets implemented once. But it's tricky--we don't want to pass State to llm.py. 

I'm OK with this for now. I'm increasingly thinking that the agent should just send _prompts_ rather than interacting with the LLM directly, which would make a check like this easier to enforce. But we'll need to do some more work before that's ready

Reply:
> But it's tricky--we don't want to pass State to llm.py.

Exactly. That's what I attempted to do first but then I realized that's far from elegant.

> I'm increasingly thinking that the agent should just send prompts rather than interacting with the LLM directly, which would make a check like this easier to enforce.

I agree with this, and any other effort to reduce the code duplicates across agents.